### PR TITLE
feat: merge mixins options

### LINKED ISSUE #261: Add support for merging mixins
A few libraries in Nuxt (vue i18n, Vuetify) uses mixins for Vue Meta.
Vue Meta for now doesn't support Mixins merging, resulting in conflicts when using both those libraries at the same time in Nuxt.

I will try to add support for mixins merging and do a pull request as offered by @Atinux 

See: https://github.com/nuxt/nuxt.js/issues/3846

### PR DESCRIPTION
**Pull request by adrienbaron**
Closes #261

### CODE REVIEW

File: src/shared/getComponentOption.js [hunk 1]
```diff
@@ -67,3 +62,17 @@ export default function getComponentOption (opts, result = {}) {
...
+function mergeDataInResult (data, result, component, arrayMerge) {
+  // if option is a function, replace it with it's result
+  if (typeof data === 'function') {
+    data = data.call(component)
+  }
+
+  if (typeof data === 'object') {
+    // merge with existing options
+    return deepmerge(result, data, { arrayMerge })
+  } else {
```
Comment (manniL) [inline src/shared/getComponentOption.js]: When extracting the code into a separate function, you could also get rid of that unnecessary else ☺️

Comment (adrienbaron) [reply src/shared/getComponentOption.js]: Sure! I'll do that as soon as I'm back home 👍

**Comment by manniL** [comment]
LGTM! (cc @Atinux)

**Comment by adrienbaron** [comment]
Any news on when this PR can be merged 🙂?

**Comment by Atinux** [comment]
Released to [v1.5.4](https://github.com/declandewet/vue-meta/releases/tag/v1.5.4), thanks @adrienbaron

**Pull request merged**